backend/battles/models.py

```diff
@@ -10,6 +12,18 @@ class Battle(models.Model):
     user_opponent = models.ForeignKey(
         User, on_delete=models.CASCADE, verbose_name="Opponent", related_name="user_opponent"
     )
+    created = models.DateTimeField(default=timezone.now)
 
     def __str__(self):
         return "Battle nº " + str(self.id)
+
+
+class Team(models.Model):
+    battle = models.ForeignKey(Battle, on_delete=models.CASCADE, related_name="battle")
+    trainer = models.ForeignKey(
+        User, on_delete=models.CASCADE, verbose_name="Trainer", related_name="trainer"
+    )
+    team = models.ManyToManyField(Pokemon, verbose_name="Team")
```

Comment:
I think you might have misunderstood the usage of the `related_name` attribute. I did too when I was at the beginning of my pokebattle.

The `related_name` attribute specifies the name of the reverse relation from the User model back to your model (line 23).

If you don't specify a `related_name`, Django automatically creates one using the name of your model with the suffix `_set`, for instance, `User.team_set.all()`.

If you do specify, e.g. `related_name=teams` on the User model, `User.team_set` will still work, but the `User.teams.` syntax is cleaner and more intuitive. So for example, if you had a user object `current_user`, you could use `current_user.teams.all()` to get all instances of your `Team` model that have a relation to `current_user`, in other words, you can get all teams from that specific user. 

**P.S.:** The logic applies not only for the `trainer` field but also for the `battle` one. Let me know if you still have doubts about the `related_name` usage, [here is a link from Django docs](https://docs.djangoproject.com/en/3.0/topics/db/queries/#following-relationships-backward) but we can chat in person as well.

---

Considering that, I would suggest the change below:

```suggestion
class Team(models.Model):
    battle = models.ForeignKey(Battle, on_delete=models.CASCADE, related_name="teams")
    trainer = models.ForeignKey(
        User, on_delete=models.CASCADE, verbose_name="Trainer", related_name="teams"
    )
    team = models.ManyToManyField(Pokemon, verbose_name="Team")
```

Reply:
So this also applies to line 9 and 12 too?
`        User, on_delete=models.CASCADE, verbose_name="Creator", related_name="user_creator"
`
becomes
`        User, on_delete=models.CASCADE, verbose_name="Creator", related_name="battles"
`
and 

`        User, on_delete=models.CASCADE, verbose_name="Opponent", related_name="user_opponent"
`

becomes
`        User, on_delete=models.CASCADE, verbose_name="Opponent", related_name="battles"
`

Reply:
Thanks, i corrected these values and the same ones in Battle models too

Reply:
Very nice that you added there as well! :) I've seen that you changed to `created_battle` and` invited_to_battle`, and it's kind of a pattern to use the `related_name` in the plural, but you can do that in next opportunities.